### PR TITLE
Improve localization

### PR DESCRIPTION
**Pull request by thomasync**
- add script to improve localization (improve-localization.js)
- fix: the script used to generate the translations (format-localization.js) didn't take certain concatenations into account.
- chore: improving the french and format-localizations.js executed
- feats: add templates to localizations

### CODE REVIEW

File: app/internal_packages/composer-templates/lib/template-store.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ class TemplateStore extends MailspringStore {
     this.listenTo(Actions.deleteTemplate, this._onDeleteTemplate);
     this.listenTo(Actions.renameTemplate, this._onRenameTemplate);
 
-    const welcomeName = 'Welcome to Templates.html';
+    const welcomeName = localized('Welcome to Templates') + '.html';
     const welcomePath = path.join(__dirname, '..', 'assets', welcomeName);
```
Comment (bengotow) [inline app/internal_packages/composer-templates/lib/template-store.ts]: Hmm I think we may need to support falling back to the english version here if the localized version doesn't exist. This is a nice feature though!

File: scripts/improve-localization.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
+const fs = require('fs');
+const path = require('path');
+const readline = require('readline');
+
+const PATH_TO_LANG = path.resolve(__dirname, '..', 'app', 'lang');
+
+const getLanguage = () => {
```
Comment (bengotow) [inline scripts/improve-localization.js:7]: We really should have done this for the old find-strings.js script, but could you add a block comment to the top of this script file explaining how to call it and what it does? It looks nice overall!

**Review by bengotow — COMMENTED** [review]
Thanks for contributing this! I left two small inline comments, but the only thing I think we need to do before merging is make the template store fall back to the default english template if a localized one cannot be found.

File: app/internal_packages/composer-templates/lib/template-store.ts (hunk 1 above)
Comment (thomasync) [reply app/internal_packages/composer-templates/lib/template-store.ts]: That's right! I forgot to put the failling back, sorry. I solved this with the **_welcomeTemplate()** function, which checks whether the file exists.

File: scripts/improve-localization.js (hunk 2 above)
Comment (thomasync) [reply scripts/improve-localization.js:7]: Thanks! I've added the indications in the **LOCALIZATION.md** file to make it more natural.

**Comment by bengotow** [comment]
Thanks, these changes look great! I think the docs will help a lot. Going to go ahead and merge this!

**Comment by PeterDaveHello** [comment]
@bengotow It seems that this has broken the progress we made in #2467 😓 I've submitted a new one in #2498, and hopefully it won't be overwritten again.

File: app/lang/zh-TW.json [hunk 3]
```diff
@@ -1,925 +1,892 @@
 {
-    "\"Launch on system start\" only works in XDG-compliant desktop environments. To enable the Mailspring icon in the system tray, you may need to install libappindicator.": "「系統啟動時啟動」僅適用於符合 XDG 標準的桌面環境。要在系統託盤中啟用 Mailspring 圖示，您可能需要安裝 libappindicator。",
```
Comment (PeterDaveHello) [inline app/lang/zh-TW.json:2]: This nearly two-thousand-line diff is extremely risky and has indeed caused issues here.

**Pull request merged**